lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -190,67 +192,63 @@ static void doUsdRename(
     applyToAllPrimSpecs(prim, renameFunc);
 }
 
-void UsdUndoRenameCommand::renameRedo()
+void renameHelper(
+    PXR_NS::UsdStageWeakPtr  stage,
+    const UsdSceneItem::Ptr& ufeSrcItem,
+    const Ufe::Path&         srcPath,
+    UsdSceneItem::Ptr&       ufeDstItem,
+    const Ufe::Path&         dstPath,
+    const std::string&       newName)
 {
-    // If the new name is the same as the current name, do nothing.
-    // This is the same behavior as the Maya rename command for Maya nodes.
-    if (_newName.empty())
-        return;
-
     // get the stage's default prim path
-    auto defaultPrimPath = _stage->GetDefaultPrim().GetPath();
+    auto defaultPrimPath = stage->GetDefaultPrim().GetPath();
 
-    const Ufe::Path srcPath = _ufeSrcItem->path();
-    const Ufe::Path dstPath = srcPath.sibling(Ufe::PathComponent(_newName));
     // Note: must fetch prim again from its path because undo/redo of composite commands
     //       (or doing multiple undo and then multiple redo) can make the cached prim
     //       stale.
-    const UsdPrim srcPrim = _stage->GetPrimAtPath(_ufeSrcItem->prim().GetPath());
+    const UsdPrim srcPrim = stage->GetPrimAtPath(ufeSrcItem->prim().GetPath());
 
-    doUsdRename(_stage, srcPrim, _newName, srcPath, dstPath);
+    doUsdRename(stage, srcPrim, newName, srcPath, dstPath);
 
     // the renamed scene item is a "sibling" of its original name.
-    _ufeDstItem = createSiblingSceneItem(srcPath, _newName);
+    ufeDstItem = createSiblingSceneItem(srcPath, newName);
 
     // update stage's default prim
-    if (_ufeSrcItem->prim().GetPath() == defaultPrimPath) {
-        _stage->SetDefaultPrim(_ufeDstItem->prim());
+    if (ufeSrcItem->prim().GetPath() == defaultPrimPath) {
+        stage->SetDefaultPrim(ufeDstItem->prim());
     }
 
     // send notification to update UFE data model
-    sendNotificationToAllStageProxies(_stage, _ufeDstItem->prim(), srcPath, dstPath);
+    sendNotificationToAllStageProxies(stage, ufeDstItem->prim(), srcPath, dstPath);
 }
 
-void UsdUndoRenameCommand::renameUndo()
+} // namespace 
+
+void UsdUndoRenameCommand::renameRedo()
 {
     // If the new name is the same as the current name, do nothing.
     // This is the same behavior as the Maya rename command for Maya nodes.
     if (_newName.empty())
         return;
 
-    // get the stage's default prim path
-    auto defaultPrimPath = _stage->GetDefaultPrim().GetPath();
-
-    const Ufe::Path srcPath = _ufeDstItem->path();
-    const Ufe::Path dstPath = _ufeSrcItem->path();
-    // Note: must fetch prim again from its path because undo/redo of composite commands
-    //       (or doing multiple undo and then multiple redo) can make the cached prim
-    //       stale.
-    const UsdPrim     prim = _stage->GetPrimAtPath(_ufeDstItem->prim().GetPath());
-    const std::string newName = _ufeSrcItem->prim().GetName();
+    const Ufe::Path srcPath = _ufeSrcItem->path();
```

Comment:
I think scene item return teh path by ref, so this could be a &, to verify.

Reply:
Good point, changed to auto so we don't need to worry about return type.